File: backend/src/main/java/com/readwise/ai/modules/news/service/NewsService.java
package com.readwise.ai.modules.news.service;

import com.readwise.ai.common.dto.PageResponse;
import com.readwise.ai.modules.news.dto.ArticleDto;

public interface NewsService {
    PageResponse<ArticleDto> getNewsFeed(String category, String query, int page, int size);
    ArticleDto getArticleById(Long id);
    void syncNewsFeed(String category);
}

File: frontend/src/services/newsService.js
import api from './api';

export const newsService = {
  getNewsFeed: async (category = 'all', query = '', page = 0, size = 10) => {
    const params = new URLSearchParams();
    if (category) params.append('category', category);
    if (query) params.append('q', query);
    params.append('page', page);
    params.append('size', size);

    const response = await api.get(`/news/feed?${params.toString()}`);
    return response.data;
  },

  getArticleById: async (id) => {
    const response = await api.get(`/news/${id}`);
    return response.data;
  },

  syncNews: async (category = 'all') => {
    const response = await api.post(`/news/sync?category=${category}`);
    return response.data;
  },
};
